workflows/klebsiella_pneumoniae.nf

```diff
@@ -25,16 +25,20 @@ include { virulencefinder                           } from '../nextflow-modules/
 include { CALL_BACTERIAL_BASE                       } from '../workflows/bacterial_base.nf'
 
 workflow CALL_KLEBSIELLA_PNEUMONIAE {
+    // Create channel for sample metadata
     Channel.fromPath(params.csv)
         .splitCsv(header:true)
         .map{ row -> get_meta(row) }
-        .branch {
-            iontorrent: it[2] == "iontorrent"
-            illumina: it[2] == "illumina"
-            nanopore: it[2] == "nanopore"
-        }
+        .map{ row -> [row[0], row[2]] }
         .set{ ch_meta }
 
+    // Create channel for reads
+    Channel.fromPath(params.csv)
+        .splitCsv(header:true)
+        .map{ row -> get_meta(row) }
+        .map{ row -> [row[0], row[1]] }
```

Comment:
Think you should rather create a method that returns the desired tuple and `.tap()` to output two channels from reading one